src/tools/actors/ActorExecutionTrace.java

```diff
@@ -287,32 +301,36 @@ public static void mailboxExecuted(final Mailbox m, final Actor actor) {
           b.putShort((short) idx);
         }
 
-        if (em instanceof PromiseSendMessage) {
-          b.put(Events.PromiseMessage.id);
+        if (em instanceof PromiseSendMessage && VmSettings.PROMISE_CREATION) {
+          b.put((byte) (eventid | PROMISE_BIT));
```

Comment:
when promise tracing is turned off, the promise message is treated like a normal message